pkg/controllers/crds/crds_controller.go

```diff
@@ -212,16 +215,21 @@ func (r *Reconciler) updateCRD(ctx context.Context, req ctrl.Request) error {
 		return err
 	}
 	r.dnsName = fmt.Sprintf("%v.%v.svc", r.SvcName, r.SvcNamespace)
-	need, err := r.refreshCertIfNeeded(&secret)
-	if err != nil {
-		return err
+	refreshedCert := false
+	if r.EnableCertRenewal {
+		refreshedCert, err = r.refreshCertIfNeeded(&secret)
+		if err != nil {
+			return err
+		}
 	}
-	if need {
+	// Injects the certificates if they were refreshed or changed
+	if refreshedCert || !r.EnableCertRenewal {
```

Comment:
```suggestion
	if refreshedCert {
```

this is not needed, refreshedCert is always false if `EnableCertRenewal` is false.
Maybe consider renaming `refreshedCert` to `shouldRefresh` for clarity

Reply:
If `|| !r.EnableCertRenewal` is removed without other changes to the logic, the certificates would only be "reinjected" when they were refreshed by `r.refreshCertIfNeeded(&secret)`. But, the idea is to let certificates be refreshed from outside of this code. If the flag `EnableCertRenewal` is deactivated (false), this code should only check if the certs changed, then reinject the certs if they changed.

Reply:
I think the point of Moritz there is that it's confusing that you have a check of that value beforehand as well as here.

There is a way to combine this into a single boolean or a decision. Since it doesn't make sense to check twice if that makes sense?